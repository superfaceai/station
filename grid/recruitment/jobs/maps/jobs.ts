/* eslint-disable jest/no-export */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

export const jobsTest = (
  provider: string,
  options?: RecordingProcessOptions
): void => {
  describe(`recruitment/jobs/${provider}`, () => {
    let superface: SuperfaceTest;

    describe('ListJobs', () => {
      beforeAll(() => {
        superface = new SuperfaceTest({
          profile: 'recruitment/jobs',
          provider,
          useCase: 'ListJobs',
          testInstance: expect,
        });
      });

      describe('when searching for published jobs', () => {
        it('performs correctly', async () => {
          const result = await superface.run(
            {
              input: {
                state: 'published',
              },
              testName: 'list published jobs',
            },
            options
          );

          expect(result.isOk).toBeTruthy();
          expect(result).toMatchSnapshot();
        });
      });

      describe('when searching for draft jobs', () => {
        it('performs correctly', async () => {
          const result = await superface.run(
            {
              input: {
                state: 'draft',
              },
              testName: 'list draft jobs',
            },
            options
          );

          expect(result.isOk).toBeTruthy();
          expect(result).toMatchSnapshot();
        });
      });
    });
  });
};
