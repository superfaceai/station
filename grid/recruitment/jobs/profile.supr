"""
Manage jobs
List, get and manage job positions of your company.
"""

name = "recruitment/jobs"
version = "1.0.0"

"""
List jobs
Returns a collection of your company jobs.
"""
usecase ListJobs {

  input {
    """
    State
    The state to filter jobs with.
    """
    state JobState
    """
    Page
    Pagination cursor (obtained from `previousPage` or `nextPage`).
    """
    page string
    """
    Page limit
    Maximum number of jobs returned in single page.
    """
    limit number
  }

  result {
    """
    Jobs
    List of jobs.
    """
    jobs [Job!]!
    """
    Pagination
    The pagination state.
    """
    pagination Pagination
    """
    Rate limit
    Rate limit details.
    """
    rateLimit RateLimit
  }

  error Error

  example Successful {
    input {
      state = 'published'
    }

    result {
      jobs = [ {
          id = 'JOB_ID',
          name = 'Software engineer',
          description = 'We are looking for a Software Engineer who will assist us with design, development and installation of software solutions. Your duties will include development, writing code, and documenting functionality.',
        }
      ]
    }
  }

  example Failed {
    input {
      limit = -1
    }

    error {
      title = 'Bad request'
      detail = 'Page limit must be a positive integer.'
      code = 'BadRequest'
      rateLimit = {
        bucket = 'workable'
      }
    }
  }

}

model Job {
  """
  Job ID
  Unique identifier of the job.
  """
  id! string!
  """
  Job name
  The name of the job.
  """
  name string
  """
  Job description
  The description of the job.
  """
  description string
  """
  Job state
  The state of the job.
  """
  state JobState
  """
  Department
  The name of the department that opened the job.
  """
  departmentName string
  """
  Requirements
  The requirements for the job to the candidate.
  """
  requirements [string]
  """
  Benefits
  The benefits of the job for the candidate.
  """
  benefits [string]
  """
  Job creation date
  The timestamp when was the job created in ISO8601 format.
  """
  createdAt string
}

model JobState enum {
  DRAFT = 'draft',
  PUBLISHED = 'published',
  CLOSED = 'closed',
  ARCHIVED = 'archived'
}

model Pagination {
  """
  Previous page
  Previous page cursor.
  """
  previousPage

  """
  Next page
  Next page cursor.
  """
  nextPage
}

model Error {
  """
  Error Title
  A short, human-readable summary of the problem type.
  """
  title!
  """
  Error Detail
  A human-readable explanation specific to this occurrence of the problem.
  """
  detail
  """
  Code
  Error code.
  """
  code! ErrorCode!
  """
  Rate limit
  Rate limit details.
  """
  rateLimit RateLimit
}

model RateLimit {
  """
  Bucket
  Different parts of API may have different rate limits.
  Bucket identifies to which part of API the rate limits apply.
  """
  bucket string
  """
  Total requests
  Total requests available in the time window.
  """
  totalRequests number
  """
  Remaining requests
  Remaining requests available in the time window.
  """
  remainingRequests number
  """
  Remaining requests available in percents
  Remaining requests available in the time window in percents.
  """
  remainingRequestsPercentage number
  """
  Reset timestamp
  Timestamp when the rate limits will reset (in  Unix Timestamp format). Available only for providers without rolling rate limit window.
  """
  resetTimestamp number
}

model ErrorCode enum {
  """
  Bad request
  The request was invalid or cannot be otherwise served.
  """
  BadRequest
  """
  Not found
  The profileId does not exist.
  """
  NotFound
  """
  Rate limit reached
  Returned when a request cannot be served due to the user or app reached max requests quota.
  """
  RateLimitReached
  """
  Unauthenticated
  There was a problem authenticating your request. Check that you are passing valid credentials.
  """
  Unauthenticated
  """
  Unauthorized
  The request is understood, but it has been refused or access is not allowed.
  """
  Unauthorized
  """
  Unknown error
  Returned when an unexpected error occured.
  """
  UnknownError
}
