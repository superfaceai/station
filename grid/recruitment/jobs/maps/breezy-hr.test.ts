import { jobsTest } from './jobs';

jobsTest('breezy-hr');
