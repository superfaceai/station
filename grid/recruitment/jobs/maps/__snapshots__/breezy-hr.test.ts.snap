// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recruitment/jobs/breezy-hr ListJobs when searching for draft jobs performs correctly 1`] = `
Ok {
  "value": Object {
    "jobs": Array [],
    "rateLimit": Object {
      "bucket": "breezy-hr",
      "remainingRequests": 98,
      "remainingRequestsPercentage": 98,
      "resetTimestamp": 1670400276,
      "totalRequests": 100,
    },
  },
}
`;

exports[`recruitment/jobs/breezy-hr ListJobs when searching for published jobs performs correctly 1`] = `
Ok {
  "value": Object {
    "jobs": Array [
      Object {
        "benefits": Array [],
        "departmentName": "Makers",
        "description": "<h2>About the Software Engineer position</h2>
<p>We are looking for a Software Engineer who will assist us with design, development and installation of software solutions. Your duties will include development, writing code, and documenting functionality.</p>
<p>You should be able to build high-quality, innovative and fully performing software in compliance with coding standards and technical design.</p>
<p>&nbsp;</p>
<h2>Software Engineer responsibilities are:</h2>
<ul>
	<li>
	<p>Organize full lifecycle software development</p>
	</li>
	<li>
	<p>Prepare specifications and determine operational feasibility</p>
	</li>
	<li>
	<p>Write well designed, testable, efficient code</p>
	</li>
	<li>
	<p>Prepare software verification plans and quality assurance procedures</p>
	</li>
	<li>
	<p>Organize documenting and maintenance of software functionality</p>
	</li>
	<li>
	<p>Integrate software components into a fully functional software system</p>
	</li>
	<li>
	<p>Evaluate and deploy software tools, processes and metrics</p>
	</li>
	<li>
	<p>Provide support and consulting on software systems usage</p>
	</li>
	<li>
	<p>Ensure compliance with project plans and industry standards</p>
	</li>
</ul>
<p>&nbsp;</p>
<h2>Software Engineer requirements are:</h2>
<ul>
	<li>
	<p>2+ years' experience of working on a Computer Software Engineer or Software Developer (http://link...) position</p>
	</li>
	<li>
	<p>Significant experience in developing web applications using at least one popular web framework, such as JSF, Wicket, GWT or Spring MVC</p>
	</li>
	<li>
	<p>Excellent knowledge of Relational Databases, SQL and ORM technologies (JPA2, Hibernate)</p>
	</li>
	<li>
	<p>Good experience in designing interactive applications and with test-driven development</p>
	</li>
	<li>
	<p>Good knowledge to develop software in C, C++, C#, Java or other selected languages</p>
	</li>
	<li>
	<p>Experience in documenting requirements and specifications</p>
	</li>
	<li>
	<p>Close familiarity with software engineering tools, software development methodology and release processes</p>
	</li>
	<li>
	<p>BS degree in Computer Science or Engineering</p>
	</li>
</ul>",
        "id": "83667ee523c901",
        "name": "Software engineer",
        "requirements": Array [],
      },
      Object {
        "benefits": Array [],
        "departmentName": null,
        "description": "<h2>About the Accountant&nbsp;position</h2>
<p>We are looking for a skilled Accountant to be responsible for the preparation, computing, managing, researching and analyzing of company's accounting data. We expect you to provide detailed information on company's performance, liquidity, solvency, and cash flows as well as analyze its financial position and make conclusions about it.</p>
<h2>Accountant responsibilities are:</h2>
<ul>
	<li>
	<p>Perform all accounting operations and ensure their compatibility with accounting principles and regulations</p>
	</li>
	<li>
	<p>Calculate taxes and create tax returns, balance sheet, profit/loss statements and similar documentation</p>
	</li>
	<li>
	<p>Create budget and financial forecasts</p>
	</li>
	<li>
	<p>Prepare timely and accurate financial statements</p>
	</li>
	<li>
	<p>Perform month-end and year-end close processes</p>
	</li>
	<li>
	<p>Perform audits of company's financial transactions and prepare documentation for accounting control procedures</p>
	</li>
	<li>
	<p>Retrieve, analyze and manage company's account information</p>
	</li>
	<li>
	<p>Prepare periodic reports for company's executives and top managers</p>
	</li>
	<li>
	<p>Keep corporate information confidential and develop safety measures such as random database backups</p>
	</li>
	<li>
	<p>Monitor all updates to financial policies, regulation, and legislation</p>
	</li>
</ul>
<h2>Accountant requirements are:</h2>
<ul>
	<li>
	<p>3+ years' experience of working on a Cost Accountant or other relevant position</p>
	</li>
	<li>
	<p>Profound knowledge of accounting, financial and business principles and procedures</p>
	</li>
	<li>
	<p>Excellent knowledge of accounting software</p>
	</li>
	<li>
	<p>Strong computation skills with close attention to detail</p>
	</li>
	<li>
	<p>Productivity and confidentiality</p>
	</li>
	<li>Advanced degree in Accounting; CPA or CMA certification would be a bonus</li>
</ul>",
        "id": "6a621a768bc101",
        "name": "Accountant",
        "requirements": Array [],
      },
    ],
    "rateLimit": Object {
      "bucket": "breezy-hr",
      "remainingRequests": 99,
      "remainingRequestsPercentage": 99,
      "resetTimestamp": 1670400276,
      "totalRequests": 100,
    },
  },
}
`;
